block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -186,6 +204,22 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
 
     override def nonFinalizedSet: Set[BlockHash] =
       dagSet diff finalizedBlocksSet
+
+    override def reachedAcquiescence: F[Boolean] =
+      for {
+        statesOfNotFinalizedBlocks <- nonFinalizedSet.toList.traverse(
+                                       this.lookupUnsafe(_).map(_.postStateHash)
+                                     )
+        // how many time each state is met across non finalized blocks
+        stateSeenInNonFinalizedCounter = statesOfNotFinalizedBlocks
+          .groupBy(identity)
+          .mapValues(_.size)
+
+        // if all occurrences of a state are in not finalized blocks - state is not finalized, no acquiescence
+        hasNotFinalizedState = stateSeenInNonFinalizedCounter.exists {
+          case (state, qty) => validStatesCounter(state) == qty
+        }
+      } yield !hasNotFinalizedState
```

Comment:
What will be the state in non-finalized blocks when DAG is in acquiescence state?

Can we just look at non-finalized blocks to calculate this?

Reply:
Non finalized blocks have the same state as some (one or several) finalized blocks. 

We need to be sure that state is met only in non finalized blocks. Thats why we need to have a visibility of which states are in finalized blocks. Or (as implemented) - have counter of how many times state is met, so by looking into non finalized blocks we are able to know that all occurrences are in non finalized blocks.